utils/SwiftBuildSupport.py

```diff
@@ -143,30 +143,35 @@ def get_preset_options(substitutions, preset_file_names, preset_name):
                           "': " + ", ".join(missing_opts))
 
     # Migrate 'swift-sdks' parameter to 'stdlib-deployment-targets'
-    for opt in build_script_impl_opts:
-        if opt.startswith("--swift-sdks"):
-            sdks_to_configure = opt.split("=")[1].split(";")
-            tgts = []
-            # Expand SDKs in to their deployment targets
-            from swift_build_support.targets import StdlibDeploymentTarget
-            for sdk in sdks_to_configure:
-                if sdk == "OSX":
-                    tgts += StdlibDeploymentTarget.OSX.targets
-                elif sdk == "IOS":
-                    tgts += StdlibDeploymentTarget.iOS.targets
-                elif sdk == "IOS_SIMULATOR":
-                    tgts += StdlibDeploymentTarget.iOSSimulator.targets
-                elif sdk == "TVOS":
-                    tgts += StdlibDeploymentTarget.AppleTV.targets
-                elif sdk == "TVOS_SIMULATOR":
-                    tgts += StdlibDeploymentTarget.AppleTVSimulator.targets
-                elif sdk == "WATCHOS":
-                    tgts += StdlibDeploymentTarget.AppleWatch.targets
-                elif sdk == "WATCHOS_SIMULATOR":
-                    tgts += StdlibDeploymentTarget.AppleWatchSimulator.targets
-
-            build_script_opts.append("--stdlib-deployment-targets=" +
+    try:
+        swift_sdks_opts = [opt for opt in build_script_impl_opts
+                if opt.startswith("--swift-sdks")]
+        opt = swift_sdks_opts[-1]
+        sdks_to_configure = opt.split("=")[1].split(";")
+        tgts = []
+        # Expand SDKs in to their deployment targets
+        from swift_build_support.targets import StdlibDeploymentTarget
+        for sdk in sdks_to_configure:
+            if sdk == "OSX":
+                tgts += StdlibDeploymentTarget.OSX.targets
+            elif sdk == "IOS":
+                tgts += StdlibDeploymentTarget.iOS.targets
+            elif sdk == "IOS_SIMULATOR":
+                tgts += StdlibDeploymentTarget.iOSSimulator.targets
+            elif sdk == "TVOS":
+                tgts += StdlibDeploymentTarget.AppleTV.targets
+            elif sdk == "TVOS_SIMULATOR":
+                tgts += StdlibDeploymentTarget.AppleTVSimulator.targets
+            elif sdk == "WATCHOS":
+                tgts += StdlibDeploymentTarget.AppleWatch.targets
+            elif sdk == "WATCHOS_SIMULATOR":
+                tgts += StdlibDeploymentTarget.AppleWatchSimulator.targets
+
+        build_script_opts.append("--stdlib-deployment-targets=" +
                                      " ".join([tgt.name for tgt in tgts]))
+    except IndexError:
```

Comment:
This looks like a hack that can hide real issues, can we scope it down to the problematic statement?